drivers/hmis/spec/models/hmis/hmis_data_cleanup_util_spec.rb

```diff
@@ -0,0 +1,222 @@
+      records << create(:hmis_hud_event, :skip_validate, **shared_attributes)
+      records
+    end
+
+    let!(:records_with_good_references) do
+      records = []
+      shared_attributes = {
+        data_source: hmis_ds,
+        enrollment: e1,
+        client: e1.client,
+        DateCreated: last_year,
+        DateUpdated: last_year,
+      }
+      records << create(:hmis_hud_service, **shared_attributes)
+      records << create(:hmis_income_benefit, **shared_attributes)
+      records << create(:hmis_health_and_dv, **shared_attributes)
+      records << create(:hmis_youth_education_status, **shared_attributes)
+      records << create(:hmis_employment_education, **shared_attributes)
+      records << create(:hmis_disability, **shared_attributes)
+      records << create(:hmis_hud_exit, **shared_attributes)
+      records << create(:hmis_current_living_situation, **shared_attributes)
+      records << create(:hmis_hud_assessment, **shared_attributes)
+      records << create(:hmis_assessment_question, **shared_attributes)
+      records << create(:hmis_assessment_result, **shared_attributes)
+      records << create(:hmis_hud_event, **shared_attributes)
+      records
+    end
+
+    it 'works for services' do
+      bad_service = create(:hmis_hud_service, :skip_validate, enrollment: e1, PersonalID: 'unmatched-id', data_source: hmis_ds)
+      good_service = create(:hmis_hud_service, :skip_validate, enrollment: e1, data_source: hmis_ds)
+
+      HmisDataCleanup::Util.fix_incorrect_personal_id_references!(classes: [Hmis::Hud::Service])
+      [bad_service, good_service].each(&:reload)
+      expect(bad_service.personal_id).to eq(e1.personal_id)
+      expect(bad_service.enrollment).to be_present
+      expect(good_service.enrollment).to be_present
+    end
+
+    it 'works for all record types' do
+      expect(records_with_bad_references.map(&:PersonalID).uniq).to contain_exactly('not-real')
+
+      expect do
+        HmisDataCleanup::Util.fix_incorrect_personal_id_references!
+
+        # no new records should have been created during the upsert
+      end.to change(GrdaWarehouse::Hud::Disability, :count).by(0).
+        and change(GrdaWarehouse::Hud::EmploymentEducation, :count).by(0).
+        and change(GrdaWarehouse::Hud::Exit, :count).by(0).
+        and change(GrdaWarehouse::Hud::HealthAndDv, :count).by(0).
+        and change(GrdaWarehouse::Hud::IncomeBenefit, :count).by(0).
+        and change(GrdaWarehouse::Hud::Service, :count).by(0).
+        and change(GrdaWarehouse::Hud::CurrentLivingSituation, :count).by(0).
+        and change(GrdaWarehouse::Hud::Assessment, :count).by(0).
+        and change(GrdaWarehouse::Hud::AssessmentQuestion, :count).by(0).
+        and change(GrdaWarehouse::Hud::AssessmentResult, :count).by(0).
+        and change(GrdaWarehouse::Hud::Event, :count).by(0).
+        and change(GrdaWarehouse::Hud::YouthEducationStatus, :count).by(0)
```

Comment:
Alright switch back to `import`ing records, and here we ensure that no new records were created

Reply:
I've had trouble with `not_to change` so I do change by 0.. not as readable but seems to be happy